Framework/Kernel/inc/MantidKernel/MultiThreaded.h

```diff
@@ -196,6 +201,15 @@ void AtomicOp(std::atomic<T> &f, T d, BinaryOp op) {
 
 #define PARALLEL_SECTION PRAGMA(omp section)
 
+#define PARALLEL_SET_THREADS_TO_CONFIG                                         \
+  if (Mantid::Kernel::ConfigService::Instance()                                \
+          .getValue<int>("MultiThreaded.MaxCores")                             \
+          .get_value_or(0) > 0) {                                              \
+    PARALLEL_SET_NUM_THREADS(Mantid::Kernel::ConfigService::Instance()         \
+                                 .getValue<int>("MultiThreaded.MaxCores")      \
+                                 .get());                                      \
+  }
+
```

Comment:
Can we do something like the following here, this will remove some of the duplicate code but keep the variable out of the caller's scope.

```cpp
{
  const auto max_cores = Mantid::Kernel::ConfigService::Instance().getValue<int>("MultiThreaded.MaxCores");
  if (max_cores.get_value_or(0) > 0) {
    PARALLEL_SET_NUM_THREADS(max_cores.get());
  }
}
```

Thinking about it, this code could just be a function, nothing about this requires it is a macro.

Reply:
I might be doing something wrong, but if I define it as a function: whenever one of the `MultiThreaded` macros is used more than once, the compiler complains that it has been defined more than once.

Reply:
You'll need to `inline` it as its in a header.

Reply:
I couldn't get that to work before, but it seems to work now.